### PR TITLE
Add support for using latest OSS-Fuzz corpora, add libxslt benchmark.

### PR DESCRIPTION
**Pull request by inferno-chromium**

### CODE REVIEW

**Comment by inferno-chromium** [comment]
Ready for review! Unsupported fuzzers tracked in https://github.com/google/fuzzbench/issues/762. 
Adding a benchmark is pain these days, some fuzzers always break.

File: .github/workflows/build_and_test_run_fuzzer_benchmarks.py [hunk 1]
```diff
@@ -88,9 +100,9 @@ def make_builds(benchmarks, fuzzer):
         make_target = get_make_target(fuzzer, benchmark)
         make_command = ['make', 'RUNNING_ON_CI=yes', '-j', make_target]
         print('Running command:', ' '.join(make_command))
-        result = subprocess.run(make_command, check=False)
-        if not result.returncode == 0:
+        if not run_with_retries(make_command):
```
Comment (jonathanmetzman) [inline .github/workflows/build_and_test_run_fuzzer_benchmarks.py:103]: Maybe reuse: https://github.com/google/fuzzbench/blob/master/common/retry.py

File: .github/workflows/fuzzers.yml [hunk 2]
```diff
@@ -37,7 +37,6 @@ jobs:
           - honggfuzz_qemu
           - weizz_qemu
           # Temporary variants.
-          - aflcc_no_orig_dict
```
Comment (jonathanmetzman) [inline .github/workflows/fuzzers.yml:40]: Don't understand what this removal is doing here.

File: benchmarks/libxslt_xpath/build.sh [hunk 3]
```diff
@@ -0,0 +1,74 @@
...
+        -o $OUT/$fuzzer \
+        $LIB_FUZZING_ENGINE \
+        libexslt/.libs/libexslt.a libxslt/.libs/libxslt.a \
+        ../libxml2/.libs/libxml2.a \
+        $CRYPTO_LIBS
+
+    zip -j $OUT/${fuzzer}_seed_corpus.zip tests/fuzz/seed/$fuzzer/*
+done
+
+cp tests/fuzz/xpath.dict tests/fuzz/xpath.xml $OUT/
```
Comment (jonathanmetzman) [inline benchmarks/libxslt_xpath/build.sh:73]: Don't know if I want zips commited to this repo or if we should put them in GCB bucket. Need to think about this.

File: fuzzers/eclipser/fuzzer.py [hunk 4]
```diff
@@ -65,7 +65,7 @@ def fuzz(input_corpus, output_corpus, target_binary):
         # Default is low (0.5 sec), recommended to use higher:
         # https://github.com/google/fuzzbench/issues/70#issuecomment-596060572
         '--exectimeout',
-        '2000',
+        '5000',
```
Comment (jonathanmetzman) [inline fuzzers/eclipser/fuzzer.py:68]: Is this change necessary because eclipser breaks otherwise?

File: .github/workflows/fuzzers.yml (hunk 2 above)
Comment (inferno-chromium) [reply .github/workflows/fuzzers.yml:40]: aflcc does not work across a lot of benchmarks [atleast 4-5]. While aflcc is explicitly excluded, this variant is not. I prefer to keep this temporary variant out of CI, rather than adding more blacklist for this temp variant.

File: benchmarks/libxslt_xpath/build.sh (hunk 3 above)
Comment (inferno-chromium) [reply benchmarks/libxslt_xpath/build.sh:73]: I prefer zip here as it is more clear for people looking at benchmark directory. Going to GCS folder and gsutil might appear obvious to us, but not for the common crowd.

File: fuzzers/eclipser/fuzzer.py (hunk 4 above)
Comment (inferno-chromium) [reply fuzzers/eclipser/fuzzer.py:68]: Yes right, this timeout hits with seed corpus now. Also, it will hit as we add more seed corpus, i remember seeing with libaom that i was planning to add too.

File: .github/workflows/build_and_test_run_fuzzer_benchmarks.py (hunk 1 above)
Comment (inferno-chromium) [reply .github/workflows/build_and_test_run_fuzzer_benchmarks.py]: Done!

**Comment by inferno-chromium** [comment]
Addressed comments, PTAL.

File: common/retry.py [hunk 5]
```diff
@@ -41,6 +39,8 @@ def wrap(  # pylint: disable=too-many-arguments
         exception_type=Exception,
         retry_on_false=False):
     """Retry decorator for a function."""
+    # To avoid circular dependency.
```
Comment (jonathanmetzman) [inline common/retry.py:42]: This seems to be hiding the dependency rather than avoiding it.
It's fine if this is needed because this CL didn't introduce this problem but comment is misleading IMO.

File: benchmarks/oss_fuzz_benchmark_integration.py [hunk 6]
```diff
@@ -96,6 +99,34 @@ def copy_oss_fuzz_files(project, commit_date, benchmark_dir):
...
+        full_fuzz_target = fuzz_target
+
+    corpus_backup_url = CORPUS_BACKUP_URL_FORMAT.format(
+        project=project, fuzz_target=full_fuzz_target)
+    command = ['gsutil', 'ls', corpus_backup_url]
+    result = subprocess.run(command, stdout=subprocess.PIPE)
+    if result.returncode:
+        raise Exception(f'No corpus found for project {project}.')
+
+    archive_filename = f'{fuzz_target}_seed_corpus.zip'
```
Comment (jonathanmetzman) [inline benchmarks/oss_fuzz_benchmark_integration.py:116]: Do you think we should call this seed corpus or something else?

File: common/retry.py (hunk 5 above)
Comment (inferno-chromium) [reply common/retry.py:42]: Yes otherwise it breaks CI. Since we need to use logs in retry, i dont see any other way to break the circular deps. This is hacky, but no other way.

File: benchmarks/oss_fuzz_benchmark_integration.py (hunk 6 above)
Comment (inferno-chromium) [reply benchmarks/oss_fuzz_benchmark_integration.py]: Done, made it oss_fuzz_corpus.zip

**Comment by inferno-chromium** [comment]
Done, addressed comments. PTAL.

File: benchmarks/libxslt_xpath/build.sh (hunk 3 above)
Comment (inferno-chromium) [reply benchmarks/libxslt_xpath/build.sh:73]: I will switch it to GCS if this explodes things, right now we are using simple benchmarks with 10-20mb benchmarks.

Comment (jonathanmetzman) [reply benchmarks/libxslt_xpath/build.sh:73]: Yes but the annoying thing is that unless we clean up the history corpora will take up space forever.

**Review by jonathanmetzman — APPROVED** [review]
LGTM for now.
